modules/core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/TypeService.java

```diff
@@ -36,107 +31,18 @@
  */
 public class TypeService extends DefaultComponent {
 
-    private static final String XP_SCHEMA = "schema";
+    /** @since 11.5 */
+    public static final String COMPONENT_NAME = "org.nuxeo.ecm.core.schema.TypeService";
 
-    private static final String XP_DOCTYPE = "doctype";
+    protected static final String XP_SCHEMA = "schema";
 
-    private static final String XP_CONFIGURATION = "configuration";
+    /** @since 11.5 */
+    public static final String XP_DOCTYPE = "doctype";
 
-    private static final String XP_DEPRECATION = "deprecation";
+    private static final String XP_CONFIGURATION = "configuration";
```

Comment:
I'll change the sort, and about the choice:
- XP_DOCTYPE (and COMPONENT_NAME) are public because of interactions with higher-level TypeService
- XP_SCHEMA is protected because of compat code to handle deprecated PropertyDeprecationDescriptor contributions (moved to the schema XP)
- XP_CONFIGURATION is private because only looked up at component start